### ISSUE TITLE
`Controller{Installation,Registration}` is not reconciled when `Seed` spec is updated

### ISSUE THREAD

**Issue by ialidzhikov** [gardener/gardener#7732]
**How to categorize this issue?**

/area quality
/kind enhancement

**What happened**:
The controllerinstallation controller is passing the following values for the extension chart - ref https://github.com/gardener/gardener/blob/9a2c97ef73768342f53a1d3db84de69c8723bc12/pkg/gardenlet/controller/controllerinstallation/controllerinstallation/reconciler.go#L188-L215. It is passing many values depending on the Seed spec and it is passing the Seed spec itself. There are extension charts that rely on the Seed spec or values coming from the Seed spec. Example - https://github.com/ialidzhikov/gardener-extension-provider-aws/blob/4793018f4a43581bdbb8988f47e3e040dfd1f004/charts/gardener-extension-provider-aws/templates/service.yaml#L15

Updating the Seed spec does not trigger reconciliation for the Controller{Installation,Registration}. See https://github.com/gardener/gardener-extension-provider-aws/pull/703#discussion_r1138688247

**What you expected to happen**:
Updating the Seed spec to trigger reconciliation for the Controller{Installation,Registration}

**How to reproduce it (as minimally and precisely as possible)**:
See https://github.com/gardener/gardener-extension-provider-aws/pull/703#discussion_r1138688247

**Anything else we need to know?**:

**Environment**:

- Gardener version: v1.67.1
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

**Comment by rfranzke** [comment]
The `ControllerRegistration` controller watches `Seed` resources here https://github.com/gardener/gardener/blob/aada4de4539a102021be7a4b309356e37b002fe7/pkg/controllermanager/controller/controllerregistration/seed/add.go#L52-L62

However, its `SeedPredicate` only returns `true` for certain changes: https://github.com/gardener/gardener/blob/aada4de4539a102021be7a4b309356e37b002fe7/pkg/controllermanager/controller/controllerregistration/seed/add.go#L111-L130

If it would also return `true` for other relevant changes, the reconciler already maintains the `resourceVersion` of the `Seed` in the `spec` of the created `ControllerInstallation` https://github.com/gardener/gardener/blob/aada4de4539a102021be7a4b309356e37b002fe7/pkg/controllermanager/controller/controllerregistration/seed/reconciler.go#L441-L453

In short, I think the only thing that needs to be "fixed"/extended is the `SeedPredicate`.
